src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,142 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] elements;
+    private int size;
+    private int start = 10;
```

Comment:
The initial capacity is hardcoded as `10`. Consider defining it as a constant, such as `private static final int INITIAL_CAPACITY = 10;`, to improve readability and maintainability.